pdbufr/bufr_structure.py

```diff
@@ -135,7 +136,58 @@ def wmo_station_id_from_bufr(observation, prefix, keys):
     block_number = int(observation[prefix + keys[0]])
     station_number = int(observation[prefix + keys[1]])
     return block_number * 1000 + station_number
-
+    
+def wmo_station_position_from_bufr(observation, prefix, keys):
+    # type: (T.Dict[str, T.Any], str, T.List[str]) -> Point
+    longitude = float(observation[prefix + keys[0]]) # easting (X)
+    latitude = float(observation[prefix + keys[1]]) # northing (Y)
+    heightOfStationGroundAboveMeanSeaLevel = float(observation.get(prefix + keys[2], 0.0))
+    return Point([longitude,latitude,heightOfStationGroundAboveMeanSeaLevel]) 
+
+def CRS_from_bufr(observation, prefix, keys):
+    # type: (T.Dict[str, T.Any], str, T.List[str]) -> str
+    bufr_CRS = int(observation.get(prefix + keys[0], 0))
+    if bufr_CRS == 0:
+        CRS = "EPSG:4326" # WGS84
+    elif bufr_CRS == 1:
+        CRS = "EPSG:4258" # ETRS89
+    elif bufr_CRS == 2:
+        CRS = "EPSG:4269" # NAD83
+    elif bufr_CRS == 3:
+        CRS = "EPSG:4314" # DHDN
+    elif bufr_CRS == 4:
+        """
+        Ellipsoidal datum using the International Reference Meridian and the International
+        Reference Pole as the prime meridian and prime pole, respectively, and the origin
+        of the International Terrestrial Reference System (ITRS) (see Note 2). The
+        International Reference Meridian, International Reference Pole and ITRS are
+        maintained by the International Earth Rotation and Reference Systems
+        Service (IERS)
+        
+        (2) When Code figure 4 is used to specify a custom coordinate reference system, the ellipsoidal
+            datum shall be an oblate ellipsoid of revolution, where the major axis is uniplanar with the
+            equatorial plane and the minor axis traverses the prime meridian towards the prime pole.
+            North corresponds to the direction from the Equator to the prime pole. East corresponds to
+            the counterclockwise direction from the prime meridian as viewed from above the North Pole.
+            In this case, the semi-major and semi-minor axes must be specified (e.g. by descriptors
+            0 01 152 and 0 01 153).
+        """
+        # TODO: get ellipsoid axes from descriptors 001152 and 001153 and create an CRS using pyproj.crs.CRS.from_cf(...)
+        CRS = eccodes.CODES_MISSING_LONG
+    elif bufr_CRS == 5:
+        """
+        Earth-centred, Earth-fixed (ECEF) coordinate system or Earth-centred rotational
+        (ECR) system. This is a right-handed Cartesian coordinate system (X, Y, Z)
+        rotating with the Earth. The origin is defined by the centre of mass of the Earth.
+        (Footnote (5) of class 27 does not apply if ECEF coordinates are specified.)
+        """
+        # TODO: create an CRS using pyproj.crs.CRS.from_cf(...)
+        CRS = eccodes.CODES_MISSING_LONG
```

Comment:
This looks wrong.

Reply:
Yes, I decided to set it to None and included an exception in bufr_read